client/src/main/java/io/atomix/copycat/client/session/ClientSessionState.java

```diff
@@ -101,10 +109,28 @@ public long getSessionId() {
    * @return The session state.
    */
   public ClientSessionState setState(Session.State state) {
-    if (this.state != state) {
-      this.state = state;
-      changeListeners.forEach(l -> l.accept(state));
+    if (state != Session.State.UNSTABLE) {
+      if (this.state != state) {
+        return setStateAndCallListeners(state);
+      }
+    } else {
+      if (this.state == Session.State.UNSTABLE) {
+        if (unstabilityTimeout > 0 && (System.currentTimeMillis() - unstableSince) > unstabilityTimeout) {
+          return setStateAndCallListeners(Session.State.STALE);
+        }
+      } else if (this.state != Session.State.STALE) {
+        unstableSince = System.currentTimeMillis();
+        return setStateAndCallListeners(state);
+      }
     }
+
+    return this;
+  }
+
+  private ClientSessionState setStateAndCallListeners(Session.State state)
+  {
```

Comment:
done,

thatz my current intellij formatter , different projects follow different styles :)
I'll need to setup separate formatter for copycat ..... do you guys use intellij ? if yes, can you guys export the formatting spec and put it somewhere?